File: library/src/test/java/com/github/sundeepk/compactcalendarview/CompactCalendarHelper.java
package com.github.sundeepk.compactcalendarview;

import android.graphics.Color;

import com.github.sundeepk.compactcalendarview.domain.Event;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.Calendar;
import java.util.Date;
import java.util.HashMap;
import java.util.List;
import java.util.Locale;
import java.util.Map;

public class CompactCalendarHelper {

    public static List<Event> getOneEventPerDayForMonth(int start, int days, long timeStamp) {
        return getOneEventPerDayForMonth(start, days, timeStamp, Color.BLUE);
    }

    //generate one event per a day for a month
    public static List<Event> getOneEventPerDayForMonth(int start, int days, long timeStamp, int color) {
        Calendar currentCalendar = Calendar.getInstance(Locale.getDefault());
        List<Event> events = new ArrayList<>();
        for(int i = start; i < days; i++){
            setDateTime(timeStamp, currentCalendar, i);
            events.add(new Event(color, currentCalendar.getTimeInMillis()));
        }
        return events;
    }

    public static List<Events> getEvents(int start, int days, long timeStamp) {
        Calendar currentCalendar = Calendar.getInstance(Locale.getDefault());
        List<Events> events = new ArrayList<>();
        for(int i = start; i < days; i++){
            setDateTime(timeStamp, currentCalendar, i);
            List<Event> eventList = new ArrayList<>();
            eventList.add(new Event(Color.BLUE, currentCalendar.getTimeInMillis()));
            Events eventsObject = new Events(currentCalendar.getTimeInMillis(), eventList);
            events.add(eventsObject);
        }
        return events;
    }


    public static List<Events> getDayEventWith2EventsPerDay(int start, int days, long timeStamp) {
        Calendar currentCalendar = Calendar.getInstance(Locale.getDefault());
        List<Events> events = new ArrayList<>();
        for(int i = start; i < days; i++){
            setDateTime(timeStamp, currentCalendar, i);
            List<Event> eventList = new ArrayList<>();
            eventList.add(new Event(Color.BLUE, currentCalendar.getTimeInMillis()));
            eventList.add(new Event(Color.RED, currentCalendar.getTimeInMillis() + 3600 * 1000));
            Events eventsObject = new Events(currentCalendar.getTimeInMillis(), eventList);
            events.add(eventsObject);
        }
        return events;
    }

    public static List<Events> getDayEventWithMultipleEventsPerDay(int start, int days, long timeStamp) {
        Calendar currentCalendar = Calendar.getInstance(Locale.getDefault());
        List<Events> events = new ArrayList<>();
        for(int i = start; i < days; i++){
            setDateTime(timeStamp, currentCalendar, i);
            List<Event> eventsList = Arrays.asList(new Event(Color.BLUE, currentCalendar.getTimeInMillis()),
                    new Event(Color.RED, currentCalendar.getTimeInMillis() + 3600 * 1000),
                    new Event(Color.RED, currentCalendar.getTimeInMillis() + (3600 * 2) * 1000),
                    new Event(Color.RED, currentCalendar.getTimeInMillis() + (3600 * 3) * 1000));
            Events eventsObject = new Events(currentCalendar.getTimeInMillis(), eventsList);
            events.add(eventsObject);
        }
        return events;
    }

    public static Map<Long, List<Event>> getMultipleEventsForEachDayAsMap(int start, int days, long timeStamp) {
        Calendar currentCalendar = Calendar.getInstance(Locale.getDefault());
        Map<Long, List<Event>> epochMillisToEvents = new HashMap<>();
        for(int i = start; i < days; i++){
            setDateTime(timeStamp, currentCalendar, i);
            List<Event> eventList = new ArrayList<>();
            List<Event> events = Arrays.asList(new Event(Color.BLUE, currentCalendar.getTimeInMillis()),
                    new Event(Color.RED, currentCalendar.getTimeInMillis() + 3600 * 1000),
                    new Event(Color.RED, currentCalendar.getTimeInMillis() + (3600 * 2) * 1000),
                    new Event(Color.RED, currentCalendar.getTimeInMillis() + (3600 * 3) * 1000));
            eventList.addAll(events);
            epochMillisToEvents.put(currentCalendar.getTimeInMillis(), eventList);
        }
        return epochMillisToEvents;
    }

    public static void setDateTime(long timeStamp, Calendar currentCalendar, int i) {
        currentCalendar.setTimeInMillis(timeStamp);
        currentCalendar.set(Calendar.DATE, 1);
        currentCalendar.set(Calendar.HOUR_OF_DAY, 0);
        currentCalendar.set(Calendar.MINUTE, 0);
        currentCalendar.set(Calendar.SECOND, 0);
        currentCalendar.set(Calendar.MILLISECOND, 0);
        currentCalendar.add(Calendar.DATE, i);
    }

    public static long setTimeToMidnightAndGet(Calendar cal, long epoch) {
        cal.setTime(new Date(epoch));
        cal.set(Calendar.HOUR_OF_DAY, 0);
        cal.set(Calendar.MINUTE, 0);
        cal.set(Calendar.SECOND, 0);
        cal.set(Calendar.MILLISECOND, 0);
        return cal.getTimeInMillis();
    }
    
}

File: library/src/test/java/com/github/sundeepk/compactcalendarview/CompactCalendarHelper.kt
package com.github.sundeepk.compactcalendarview

import android.graphics.Color
import com.github.sundeepk.compactcalendarview.domain.Event
import java.util.*

object CompactCalendarHelper {
    fun getOneEventPerDayForMonth(start: Int, days: Int, timeStamp: Long): MutableList<Event> {
        return getOneEventPerDayForMonth(start, days, timeStamp, Color.BLUE)
    }

    //generate one event per a day for a month
    fun getOneEventPerDayForMonth(start: Int, days: Int, timeStamp: Long, color: Int): MutableList<Event> {
        val currentCalender = Calendar.getInstance(Locale.getDefault())
        val events: MutableList<Event> = ArrayList()
        for (i in start until days) {
            setDateTime(timeStamp, currentCalender, i)
            events.add(Event(color, currentCalender.timeInMillis))
        }
        return events
    }

    fun getEvents(start: Int, days: Int, timeStamp: Long): List<Events> {
        val currentCalender = Calendar.getInstance(Locale.getDefault())
        val events: MutableList<Events> = ArrayList()
        for (i in start until days) {
            setDateTime(timeStamp, currentCalender, i)
            val eventList: MutableList<Event> = ArrayList()
            eventList.add(Event(Color.BLUE, currentCalender.timeInMillis))
            val eventsObject = Events(currentCalender.timeInMillis, eventList)
            events.add(eventsObject)
        }
        return events
    }

    fun getDayEventWith2EventsPerDay(start: Int, days: Int, timeStamp: Long): List<Events> {
        val currentCalender = Calendar.getInstance(Locale.getDefault())
        val events: MutableList<Events> = ArrayList()
        for (i in start until days) {
            setDateTime(timeStamp, currentCalender, i)
            val eventList: MutableList<Event> = ArrayList()
            eventList.add(Event(Color.BLUE, currentCalender.timeInMillis))
            eventList.add(Event(Color.RED, currentCalender.timeInMillis + 3600 * 1000))
            val eventsObject = Events(currentCalender.timeInMillis, eventList)
            events.add(eventsObject)
        }
        return events
    }

    fun getDayEventWithMultipleEventsPerDay(start: Int, days: Int, timeStamp: Long): List<Events> {
        val currentCalender = Calendar.getInstance(Locale.getDefault())
        val events: MutableList<Events> = ArrayList()
        for (i in start until days) {
            setDateTime(timeStamp, currentCalender, i)
            val eventsList = mutableListOf(Event(Color.BLUE, currentCalender.timeInMillis),
                    Event(Color.RED, currentCalender.timeInMillis + 3600 * 1000),
                    Event(Color.RED, currentCalender.timeInMillis + 3600 * 2 * 1000),
                    Event(Color.RED, currentCalender.timeInMillis + 3600 * 3 * 1000))
            val eventsObject = Events(currentCalender.timeInMillis, eventsList)
            events.add(eventsObject)
        }
        return events
    }

    fun getMultipleEventsForEachDayAsMap(start: Int, days: Int, timeStamp: Long): Map<Long, List<Event>> {
        val currentCalender = Calendar.getInstance(Locale.getDefault())
        val epochMillisToEvents: MutableMap<Long, List<Event>> = HashMap()
        for (i in start until days) {
            setDateTime(timeStamp, currentCalender, i)
            val eventList: MutableList<Event> = ArrayList()
            val events = mutableListOf(Event(Color.BLUE, currentCalender.timeInMillis),
                    Event(Color.RED, currentCalender.timeInMillis + 3600 * 1000),
                    Event(Color.RED, currentCalender.timeInMillis + 3600 * 2 * 1000),
                    Event(Color.RED, currentCalender.timeInMillis + 3600 * 3 * 1000))
            eventList.addAll(events)
            epochMillisToEvents[currentCalender.timeInMillis] = eventList
        }
        return epochMillisToEvents
    }

    fun setDateTime(timeStamp: Long, currentCalender: Calendar, i: Int) {
        currentCalender.timeInMillis = timeStamp
        currentCalender[Calendar.DATE] = 1
        currentCalender[Calendar.HOUR_OF_DAY] = 0
        currentCalender[Calendar.MINUTE] = 0
        currentCalender[Calendar.SECOND] = 0
        currentCalender[Calendar.MILLISECOND] = 0
        currentCalender.add(Calendar.DATE, i)
    }

    fun setTimeToMidnightAndGet(cal: Calendar, epoch: Long): Long {
        cal.time = Date(epoch)
        cal[Calendar.HOUR_OF_DAY] = 0
        cal[Calendar.MINUTE] = 0
        cal[Calendar.SECOND] = 0
        cal[Calendar.MILLISECOND] = 0
        return cal.timeInMillis
    }
}
